src/playground/sample_data_tooling/sample_data_commons/utils.py

```diff
@@ -0,0 +1,41 @@
+"""
+Copyright OpenSearch Contributors
+SPDX-License-Identifier: Apache-2.0
+
+utils.py contains the small utility functions needed throughout the tool
+
+Functions:
+    - validate_filename(): Function that validates the filename is a json, ndjson, or csv file
+"""
+
+from shutil import copyfileobj
+import gzip
+
+def validate_filename(filename):
+    """
+    Validates that the filename is a string and that the file is a json, ndjson, or csv file
+
+    Arguments:
+        - filename: The object to validate
+
+    Raises:
+        - TypeError: filename should be a string
+        - ValueError: filename must be a .json, .ndjson, or .csv file
+    """
+    if type(filename) is not str:
+        raise TypeError("filename should be a string")
+    if ".json" not in filename and ".csv" not in filename and ".ndjson" not in filename:
+        raise ValueError("filename must be a .json, .ndjson, or .csv file")
+
+def unzip_file(filename):
+    validate_filename(filename)
+    if ".json" not in filename and ".ndjson" not in filename:
```

Comment:
if function is unzip_file, why we need to check .json .ndjson. I'm confused here. what's the different between line 31 and 32.

Reply:
Yeah `unzip_file()` has been updated since this commit since this function is used in other PRs. New code does not raise errors now by design.